plugins/sonarqube-issues/src/components/SonarqubeIssues.tsx

```diff
@@ -31,20 +46,44 @@ const SonarqubeIssues: React.FC<GitIssuesProps> = ({ entityYaml }) => {
   const [issueForComment, setIssueForComment] = useState<string | null>(null);
   const [isModalOpen, setIsModalOpen] = useState(false);
   const [commentText, setCommentText] = useState("");
+  const [isCommenting, setIsCommenting] = useState(false);
+
+  const project = useMemo((): string => {
+    return entityYaml?.info?.["x-cortex-static-analysis"]?.sonarqube?.project;
+  }, [entityYaml?.info]);
+
+  useEffect(() => {
+    if (!context?.apiBaseUrl) {
+      return;
+    }
+    const fetchPluginConfig = async (): Promise<void> => {
+      let newBaseUrl = "https://sonarcloud.io";
+      try {
+        const response = await fetch(
+          `${context.apiBaseUrl}/catalog/sonarqube-plugin-config/openapi`
+        );
+        const data = await response.json();
+        newBaseUrl = data.info["x-cortex-definition"]["sonarqube-url"];
```

Comment:
This is slightly bugged, we should be pulling newBaseUrl as an optional value *before* attempting the fetch. I am running SonarQube locally and exposing a URL with ngrok, and I noticed that when I use a custom URL (check out my config in the justinreock-sandbox tenant), the service never actually gets hit. That's because if the endpoint isn't sonarcloud.io and since you are using await, the exception will trigger on the event loop before newBaseUrl has a chance to populate. I tested by adding some console logging:

`        newBaseUrl = data.info["x-cortex-definition"]["sonarqube-url"];
        console.log("Switching to updated URL" + newBaseUrl);
      } catch (e) {}
      setBaseUrl(newBaseUrl);
      console.error("Trying new base URL" + newBaseUrl);`

And you can see in the console that the variable is undefined:

<img width="569" alt="image" src="https://github.com/user-attachments/assets/866bb365-0aef-4f16-92ab-b76a6d7d7312">

Despite the plugin configured as:

<img width="824" alt="image" src="https://github.com/user-attachments/assets/34059048-3e0b-49de-9c42-c7502f6f103f">

Reply:
i am thinking to do like this
```
  useEffect(() => {
    if (!context?.apiBaseUrl) {
      return;
    }
    const fetchPluginConfig = async (): Promise<void> => {
      console.log("Fetching plugin config");
      let newBaseUrl = "https://sonarcloud.io";
      try {
        const response = await fetch(
          `${context.apiBaseUrl}/catalog/sonarqube-plugin-config/openapi`
        );
        const data = await response.json();
        const baseUrlFromEntity = data.info["x-cortex-definition"]["sonarqube-url"];
        console.log("baseUrlFromEntity is ", baseUrlFromEntity);
        if (baseUrlFromEntity) {
          newBaseUrl = baseUrlFromEntity;
        }
      } catch (e) {
        console.log("Failed to fetch plugin config", e);
      }
      console.log("Setting base url to ", newBaseUrl);
      setBaseUrl(newBaseUrl);
    };
    void fetchPluginConfig();
  }, [context?.apiBaseUrl]);
```

so that if the plugin-config can't be fetched, I get
<img width="510" alt="image" src="https://github.com/user-attachments/assets/5492b39e-b7ac-457e-8e0b-92ae499edcd5">

if the plugin-config exists but doesn't have the key (your situation) I get
<img width="512" alt="image" src="https://github.com/user-attachments/assets/976c84e9-5084-43d4-b04b-8703be910a10">

and if the plugin-config exists and has the key, I get
<img width="500" alt="image" src="https://github.com/user-attachments/assets/aeb8320e-ee38-4417-834d-3db1f396e797">

Reply:
Aha, and the reason I was missing a key is because the docs say to use sonarqube-api-url, but the code is looking for sonarqube-url: 

<img width="646" alt="image" src="https://github.com/user-attachments/assets/df57197e-8d19-4d52-b1ff-fd9201359011">

Will comment in the readme!